PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -355,31 +366,100 @@ function loadSettings(localSettingsPath : string | undefined) {
     return loadProperties(localSettingsFile);
 }
 
-function loadProperties(localSettingsFile : string) {
+function loadProperties(localSettingsFile : string) : ReadOnlyProperties | null {
     let input: string;
     try {
         input = fs.readFileSync(localSettingsFile, 'utf-8');
     } catch (err) {
-        tl.error('Error while reading local settings file.');
+        tl.warning('Error while reading local settings file.');
         return null;
     }
 
     try {
-        let props = dp.parse(input, false);
+        let props = dp.parse(input, false) as ReadOnlyProperties;
         if (!props || Object.keys(props).length === 0) {
             tl.warning('No local settings properties loaded.');
         }
         tl.debug('Local settings properties: ' + JSON.stringify(props));
         return props;
     } catch (err) {
-        tl.error('Error while parsing local settings file.');
+        tl.warning('Error while parsing local settings file.');
         return null;
     }
 }
 
+function getDtpBaseUrl(settings : ReadOnlyProperties) : string {
+    const dtpUrl = settings['dtp.url'];
+    if (!isNullOrWhitespace(dtpUrl)) {
+        try {
+            tl.debug('dtp.url: ' + dtpUrl);
+            return new URL(dtpUrl).href;
+        } catch (err) {
+            tl.warning('Invalid dtp.url value in local settings file.');
+            return '';
+        }
+    }
+
+    const dtpServer = settings['dtp.server'];
+    if (isNullOrWhitespace(dtpServer)) {
+        tl.warning('Both dtp.url and dtp.server are not specified in local settings file.');
+        return '';
+    }
+    tl.debug('dtp.server: ' + dtpServer);
+
+    let dtpBaseUrl : URL;
+    try {
+        dtpBaseUrl = new URL('https://' + dtpServer);
+    } catch (err) {
+        tl.warning('Invalid dtp.server value in local settings file.');
+        return '';
+    }
+
+    const dtpPort = settings['dtp.port'];
+    if (isValidPort(parseInt(dtpPort))) {
+        tl.debug('dtp.port: ' + dtpPort);
+        dtpBaseUrl.port = dtpPort;
+    } else {
+        tl.warning('Invalid dtp.port value in local settings file.');
+    }
+
+    const dtpContextPath = settings['dtp.context.path'];
+    if (!isNullOrWhitespace(dtpContextPath)) {
+        tl.debug('dtp.context.path: ' + dtpContextPath);
+        dtpBaseUrl.pathname = dtpContextPath;
+    }
+    
+    return dtpBaseUrl.href;
+}
+
+function checkDtpSettings(url : string, username : string, password : string) {
+    if (isNullOrWhitespace(url)) {
+        return false;
+    }
+    tl.debug('The URL to DTP server: ' + url);
+
+    if (isNullOrWhitespace(username)) {
+        tl.warning('dtp.user is not specified in local settings file.');
+        return false;
+    }
+    tl.debug('dtp.user: ' + username);
+
+    if (isNullOrWhitespace(password)) {
+        tl.warning('dtp.password is not specified in local settings file.');
+        return false;
+    }
+    tl.debug('dtp.password: ' + password);
```

Comment:
Considering security, may we should not show password in debug message.

Reply:
Yes, I removed many debug messages. Maybe it's not necessary to print out values when it's successfully loaded. People who configured the pipeline should have access to the settings file.